File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {

        int start = 0;
        int mid = 0;
        int end = nums.length -1;

        while(mid <= end){
            if(nums[mid] == 0){
                swap(nums,start, mid);
                mid++;
                start++;
            }else if (nums[mid] == 1){
                mid++;
            }else{
                swap(nums,mid, end);
                end--;
            }
        }
        
    }

    private void swap(int[] nums, int left, int right){
        int temp = nums[left];
         nums[left] = nums[right];
         nums[right] = temp;
    } 
}

File: 0075-sort-colors/0075-sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function (nums) {
  let n = nums.length;

  let start = 0;
  let middle = 0;
  let end = n - 1;

  while (middle <= end) {
    if (nums[middle] == 1) {
      middle++;
    } else if (nums[middle] == 0) {
      let temp = nums[middle];
      nums[middle] = nums[start];
      nums[start] = temp;
      start++;
      middle++;
    } else {
      let temp = nums[middle];
      nums[middle] = nums[end];
      nums[end] = temp;
      end--;
    }
  }
};
